### PR TITLE
[arp_responder]: Add IPv6 support

### PR DESCRIPTION
**Pull request by theasianpianist**
* Filter for both ARP and Neighbor Solicitation messages
* Create Neighbor Advertisement messages as configured to reply to NS
messages

Signed-off-by: Lawrence Lee <lawlee@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
ARP responder only supports IPv4 ARP requests/replies

#### How did you do it?
Add Neighbor Solicitation messages to the BPF query used to filter incoming packets
Distinguish ARP vs NDP messages based on length
Construct appropriate NA messages to reply to NS messages

#### How did you verify/test it?
Send NA messages from a machine to arp_responder running on a PTF, and verify that NS messages were sent back.
Run a set of nightly test cases to verify that existing ARP functionality was not impacted.

### CODE REVIEW

File: tests/scripts/arp_responder.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def __del__(self):
             self.socket.close()
 
     def bind(self):
-        self.socket = scapy2.conf.L2listen(iface=self.iface, filter='arp')
+        self.socket = scapy2.conf.L2listen(iface=self.iface, filter='arp || ip6[40] = 135')
```
Comment (lguohan) [inline tests/scripts/arp_responder.py:49]: make 135 as a MACRO?

**Review by lguohan — APPROVED** [review]

File: tests/scripts/arp_responder.py (hunk 1 above)
Comment (theasianpianist) [reply tests/scripts/arp_responder.py]: Done

**Comment by theasianpianist** [comment]
retest this please

**Comment by theasianpianist** [comment]
retest this please

File: tests/scripts/arp_responder.py [hunk 2]
```diff
@@ -121,6 +127,24 @@ def action(self, interface):
...
+        if target_ip_str in self.ip_sets[interface.name()]:
+            remote_ip_str = socket.inet_ntop(socket.AF_INET6, remote_ip)
+            neigh_adv_pkt = self.generate_neigh_adv(self.ip_sets[interface.name()][target_ip_str], remote_mac, target_ip_str, remote_ip_str)
+            interface.send(neigh_adv_pkt)
+
+        return
+        
+    def extract_ndp_info(self, data):
+        remote_mac = data[6:12]
+        remote_ip = data[22:38]
```
Comment (prsunny) [inline tests/scripts/arp_responder.py:143]: This would depend on whether Vlan tag is present, correct?

File: tests/scripts/arp_responder.py [hunk 3]
```diff
@@ -15,6 +15,8 @@
 scapy2.conf.use_pcap=True
 import scapy.arch.pcapdnet
 
+NEIGH_ADV_ICMP_MSG_TYPE = 135
```
Comment (prsunny) [inline tests/scripts/arp_responder.py:18]: Could you please rename it to NEIGH_SOLICIT_..?

**Pull request merged**